processor/src/test/java/org/mapstruct/ap/test/nestedbeans/mixed/AutomappingAndNestedTest.java

```diff
@@ -0,0 +1,281 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.nestedbeans.mixed;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.FishDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.FishTankDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.FishTankWithNestedDocumentDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.MaterialDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.MaterialTypeDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.OrnamentDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.WaterPlantDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.WaterQualityDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.WaterQualityOrganisationDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.WaterQualityReportDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.WaterQualityWithDocumentDto;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.Fish;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.FishTank;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.Interior;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.MaterialType;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.Ornament;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.WaterPlant;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.WaterQuality;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.WaterQualityReport;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+import org.mapstruct.ap.testutil.runner.GeneratedSource;
+
+/**
+ *
+ * @author Sjaak Derksen
+ */
+@WithClasses({
+    FishDto.class,
+    FishTankDto.class,
+    WaterPlantDto.class,
+    MaterialDto.class,
+    MaterialTypeDto.class,
+    OrnamentDto.class,
+    WaterQualityDto.class,
+    WaterQualityReportDto.class,
+    WaterQualityOrganisationDto.class,
+    Fish.class,
+    FishTank.class,
+    WaterPlant.class,
+    MaterialType.class,
+    Interior.class,
+    Ornament.class,
+    WaterQuality.class,
+    WaterQualityReport.class,
+    FishTankWithNestedDocumentDto.class,
+    WaterQualityWithDocumentDto.class,
+    FishTankMapper.class,
+    FishTankMapperConstant.class,
+    FishTankMapperExpression.class,
+    FishTankMapperWithDocument.class
```

Comment:
I guess at some point adding all classes of a package (ignoring *Test classes) would make things more readable, right? We should add that to our test framework...